openstates/az/bills.py

```diff
@@ -139,17 +139,18 @@ def actor_from_action(self, bill, action):
         for key in action_map:
             if key in action:
                 if action_map[key] == 'chamber':
-                    return bill['chamber']
+                    return self_chamber
                 else:
                     return action_map[key]
 
-    def scrape(self, chamber, session):
-        try:
-            session_id = self.get_session_id(session)
-        except KeyError:
-            raise NoDataForPeriod(session)
-
-        #Get the bills page to start the session
+    def scrape(self, chamber=None, session=None):
+        if not session:
+            session = self.latest_session()
+            self.info('no session specified, using %s', session)
+            session_id = session_metadata.session_id_meta_data[session]
```

Comment:
if we drop the `else` and If `session='something'` then we will get the id of latest session only.